src/runtime/job.cpp

```diff
@@ -744,8 +755,57 @@ static void launch(JobTable *jobtable) {
     jobtable->imp->active += task.job->threads();
     jobtable->imp->phys_active += task.job->memory();
 
-    std::shared_ptr<JobEntry> entry =
-        std::make_shared<JobEntry>(jobtable->imp.get(), std::move(task.job));
+    int fd_out = status_get_fd(task.job->stream_out.c_str());
+    int color_out = status_get_colour(task.job->stream_out.c_str());
+    int fd_err = status_get_fd(task.job->stream_err.c_str());
+    int color_err = status_get_colour(task.job->stream_err.c_str());
+
+    // Make the raw output streams and the TermInfoBufs
+    // that jobs will use. We make one TermInfoBuf per
+    // file descriptor that we're outputting to.
+    // TODO: We could add file tee-ing here as well
+    if (!jobtable->imp->fd_bufs.count(fd_out)) {
+      std::unique_ptr<std::streambuf> fd_buf;
+      if (fd_out != -1) {
+        fd_buf = std::make_unique<FdBuf>(fd_out);
+      } else {
+        fd_buf = std::make_unique<NullBuf>();
+      }
+      jobtable->imp->term_bufs[fd_out] =
+          std::make_unique<TermInfoBuf>(fd_buf.get(), !isatty(fd_out));
+      jobtable->imp->fd_bufs[fd_out] = std::move(fd_buf);
+    }
+    if (!jobtable->imp->fd_bufs.count(fd_err)) {
+      std::unique_ptr<std::streambuf> fd_buf;
+      if (fd_err != -1) {
+        fd_buf = std::make_unique<FdBuf>(fd_err);
+      } else {
+        fd_buf = std::make_unique<NullBuf>();
+      }
+      jobtable->imp->term_bufs[fd_err] =
+          std::make_unique<TermInfoBuf>(fd_buf.get(), !isatty(fd_err));
+      jobtable->imp->fd_bufs[fd_err] = std::move(fd_buf);
+    }
+
+    // Now we need to figure out if we're outputting to a null buffer to avoid rendering
+    // or if we're going to output to a nicely rendered StatusBuf.
+    auto job_id_str = wcl::some(task.job->label->as_str());
+    wcl::optional<std::string>{wcl::in_place_t{}, "job_id = " + std::to_string(task.job->job)};
```

Comment:
nit: `job_id = 123` still feels a  bit too verbose. maybe `job(123)`?

Reply:
It's actually a typo that that was left in. We're using the label now instead.